File: prefab/prefab-dependency/app/src/main/java/constantin/prefab_dependency/MainActivity.java
package constantin.prefab_dependency;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'application-lib' library on application startup.
    static {
        System.loadLibrary("application-lib");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Example of a call to a native method
        TextView tv = findViewById(R.id.sample_text);
        tv.setText(stringFromJNI());
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();
}

File: unit-test/app/src/main/java/com/example/unittest/MainActivity.kt
package com.example.unittest

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.example.unittest.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.sampleText.text = "1 + 2 = " + add(1, 2).toString()
    }

    /**
     * A native method that is implemented by the 'unittest' native library,
     * which is packaged with this application.
     */
    external fun add(a: Int, b: Int): Int

    companion object {
        // Used to load the 'unittest' library on application startup.
        init {
            System.loadLibrary("unittest")
        }
    }
}